docs/howto/effectful_testing.md

```diff
@@ -0,0 +1,368 @@
+---
+id: howto_effectful_testing
+title:  "Effectful testing"
```

Comment:
How about we use verbs so that it sounds better after "How to...". Here "Test effects"